tests/vector/test_vector_env.py

```diff
@@ -112,3 +112,34 @@ def thunk():
     assert info["final_observation"] == np.array([0]) and info["final_info"] == {
         "action": 3
     }
+
+
+@pytest.mark.parametrize(
+    "vectoriser",
+    (
+        SyncVectorEnv,
+        partial(AsyncVectorEnv, shared_memory=True),
+        partial(AsyncVectorEnv, shared_memory=False),
+    ),
+    ids=["Sync", "Async with shared memory", "Async without shared memory"],
+)
+def test_reset_with_list_of_options(vectoriser):
```

Comment:
Looks good, could you add another test that on autoreset, the options will be used again

Reply:
Hmm, at the moment, autoreset doesn't reused the provided `options` nor `seeds`.

Reply:
I don't think it should reuse seeds, that would lead to invalid trainings (with basically fixed initial states). Reusing options makes a lot of sense, we should probably cache this, either on the environment side, or inside `VectorEnv` by default